examples/platforms/simulation/radio.c

```diff
@@ -843,13 +850,47 @@ void radioSendAck(void)
     // Use enh-ack for 802.15.4-2015 frames
     if (otMacFrameIsVersion2015(&sReceiveFrame))
     {
+        uint8_t  linkMetricsDataLen = 0;
+        uint8_t *dataPtr            = NULL;
+
+#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE
+        uint8_t      linkMetricsData[2];
+        otMacAddress macAddress;
+
+        otEXPECT(otMacFrameGetSrcAddr(&sReceiveFrame, &macAddress) == OT_ERROR_NONE);
+
+        if (macAddress.mType == OT_MAC_ADDRESS_TYPE_SHORT)
```

Comment:
Can't this logic be put inside a single `otLinkMetricsEnhAckGetData` function with an `otMacAddress` argument?